### ISSUE TITLE
Support spaced usernames in ssh config and also improve handling of users with LOTS of keys

### ISSUE THREAD

**Issue by calevans** [lando/lando#2031]
Setup 
 - Windows 10 Pro 1903 Build 18362.657
 - Lando v3.0.0-rc.23
 - Docker 19.03.5

.lando.yml
```
recipe: wordpress
name: slidebot
config:
  webroot: web
  php: '7.3'
  via: 'apache'
  xdebug: false
  ssl: true
  database: 'mariadb'
services:
  database:
    portforward: true
  appserver:
    build_as_root:
      - apt-get update -y
      - apt-get install sudo nano vim unrtf cron supervisor jq -y
    build:
      - /scripts/makePrompt.sh
      - /scripts/makeapi.sh
```
In building a new container, any container, I run into 2 issues.

Issue 1:
My user name on this machine is Cal Evans. `/script/sload-keys.ssh` copies that directly into `/etc/ssh/ssh_config` even though it's invalid.

Proposed resolution 1:
Detect a space in the user name. A possible solution is simply to strip the space. that may work for some. For me, stripping it and making it all lower case works.

Issue 2:
I have around 30 keys in my .ssh dir. By copying them all into a single config file, ssh runs out of attempts long before it finds the correct key.

Proposed resolution 2:
If a config file is detected in `/lando_keys`, copy that to `~/.ssh` along with all the keys. If problem #1 is resolved, at least to the point where the config file is valid - `User Cal Evans` tells me that there is garbage at ` Evans` - then ssh will work.

Alternate proposed resolution 1&2
Allow `/scripts/load-keys.sh` to be overridden. I realize that these may both be edge cases. In that case, if I could substitute my own load-keys.sh that does things the way I need them done, then I can handle fixing it myself.

Thanks for all your hard work. I do dearly love lando. :)

Cheers!
=C=

**Comment by pirog** [comment]
This is a great report @calevans! This also has a lot in common with https://github.com/lando/lando/issues/1956. So my proposal on how to address these things are 

1. Just fix the name thing, im guessing its probably not super tricky to support names with spaces or to parse them so they dont have spaces

2. The lots of ssh keys one is more tricky but my proposal is to allow users to add a `keys` key in their user level Lando override file (details on what that is here: https://docs.lando.dev/config/lando.html#override-file)

```yaml
keys:
  - id_rsa
  - awesome_key
```

This would tell lando to only use the above user keys for this app instead of trying to load all of them. I think it would also be great to add some UX so that if Lando detects the user has 10 or more keys we can warn them and link to the docs that describe the feature above.

@calevans what do you think?

**Comment by calevans** [comment]
Hi!

Yes, this would work. It's not my preferred solution but my preferred solution is only preferred because it is how *I* solved it. :)

I thought about it over the weekend and yes, this would work. The user name thing is really the only tricky part. 

Again, allowing a hook to override this at any point after the /etc/ssh has been created would solve this. I don't need Lando to figure anything out, I just need to erase the /etc/ssh config file and put mine in the proper place. Currently, I have tooling that does this and then does a test ssh into github to prove it works. 

HTH,
Thanks for Lando. :)
=C=

**Comment by calevans** [comment]
Oh, BTW, the current system seems to work fine for most users. So adding a hook, allows those of us who are edge cases to fix it ourselves. :)

I would not break what seems to be working for most. :)

Cheers!
=C=

**Comment by pirog** [comment]
lol @calevans i feel you re "preferred solution". 

totally agreed that the current solution works well for most users, i think thats why the proposal above makes sense eg "do the normal thing unless the user says to do something else"

also good point on tooling (and probably true for overrides and build steps as well) if people want to completely circumvent lando and load their own file or logic they can do that with existing lando mechanisms

**Comment by calevans** [comment]
> also good point on tooling (and probably true for overrides and build steps as well) if people want to completely circumvent lando and load their own file or logic they can do that with existing lando mechanisms, might be good to document that option for people.

If it's possible to either overwrite or circumnavigate Lando's existing key system then yes, please, my all means document that and I'll be out of your hair.

The problem does not manifest itself during my normal development cycle, the tooling I came up with fixes that. The problem arises during lando start. After the key infrastructure is copied over but before I can find a hook to hook in,  our system does some `composer` stuff. This requires my keys, which are broke. (both #1 and #2)  If there is a `post-ssh-keys` hook (there is not) or something close to that (`post-core-plugin`?) hook that I can grab, that would rock.

Thanks for your help on this.

Love Lando.

Cheers!
=C=

**Comment by pirog** [comment]
@calevans i'll document this but i think already possible to circumvent Lando by mounting the sshconfig you want into the relevant lando service at `~/.ssh/config`. 

This user level file _should_ take precedence over the system level one Lando sets at `/etc/ssh/ssh_config`

**Comment by calevans** [comment]
I've been doing this in tooling by copying it over from /lando_keys. LMK when the documentation is ready, I'd LOVE to have it be that simple. :)

Cheers! :)
=C=

**Comment by pirog** [comment]
alright @calevans this is merged into master and will show up in 3.0.0-rrc.5, relevant docs are over here
https://docs.lando.dev/config/ssh.html (in the "customizing" section)

cc @daggerhart
